quarkus/runtime/src/test/java/or/keycloak/quarkus/runtime/cli/PicocliTest.java

```diff
@@ -0,0 +1,209 @@
+            config = createConfig();
+            super.parseAndRun(cliArgs);
+        }
+
+    };
+
+    NonRunningPicocli pseudoLaunch(String... args) {
+        NonRunningPicocli nonRunningPicocli = new NonRunningPicocli();
+        nonRunningPicocli.parseAndRun(Arrays.asList(args));
+        return nonRunningPicocli;
+    }
+
+    @Test
+    public void testNegativeArgument() {
+        NonRunningPicocli nonRunningPicocli = pseudoLaunch("start-dev");
+        assertEquals(CommandLine.ExitCode.OK, nonRunningPicocli.exitCode);
+        assertEquals("1h",
+                nonRunningPicocli.config.getConfigValue("quarkus.http.ssl.certificate.reload-period").getValue());
+
+        nonRunningPicocli = pseudoLaunch("start-dev", "--https-certificates-reload-period=-1");
+        assertEquals(CommandLine.ExitCode.OK, nonRunningPicocli.exitCode);
+        assertNull(nonRunningPicocli.config.getConfigValue("quarkus.http.ssl.certificate.reload-period").getValue());
+    }
+
+    @Test
+    public void testInvalidArgumentType() {
+        NonRunningPicocli nonRunningPicocli = pseudoLaunch("start-dev", "--http-port=a");
+        assertEquals(CommandLine.ExitCode.USAGE, nonRunningPicocli.exitCode);
+        assertThat(nonRunningPicocli.getErrString(),
+                containsString("Invalid value for option '--http-port': 'a' is not an int"));
+    }
+
+    @Test
+    public void failWrongEnumValue() {
+        NonRunningPicocli nonRunningPicocli = pseudoLaunch("start-dev", "--log-console-level=wrong");
+        assertEquals(CommandLine.ExitCode.USAGE, nonRunningPicocli.exitCode);
+        assertThat(nonRunningPicocli.getErrString(), containsString(
+                "Invalid value for option '--log-console-level': wrong. Expected values are: off, fatal, error, warn, info, debug, trace, all"));
+    }
+
+    @Test
+    public void failMissingOptionValue() {
+        NonRunningPicocli nonRunningPicocli = pseudoLaunch("start-dev", "--db");
+        assertEquals(CommandLine.ExitCode.USAGE, nonRunningPicocli.exitCode);
+        assertThat(nonRunningPicocli.getErrString(), containsString(
+                "Option '--db' (vendor) expects a single value. Expected values are: dev-file, dev-mem, mariadb, mssql, mysql, oracle, postgres"));
+    }
+
+    @Test
+    public void failMultipleOptionValue() {
+        NonRunningPicocli nonRunningPicocli = pseudoLaunch("build", "--db", "mysql", "postgres");
+        assertEquals(CommandLine.ExitCode.USAGE, nonRunningPicocli.exitCode);
+        assertThat(nonRunningPicocli.getErrString(), containsString("Unknown option: 'postgres'"));
+    }
+
+    @Test
+    public void failMultipleMultiOptionValue() {
+        NonRunningPicocli nonRunningPicocli = pseudoLaunch("build", "--features", "linkedin-oauth", "account3");
+        assertEquals(CommandLine.ExitCode.USAGE, nonRunningPicocli.exitCode);
+        assertThat(nonRunningPicocli.getErrString(), containsString("Unknown option: 'account3'"));
+    }
```

Comment:
Just wanted to highlight this. It is essentially a step back that we no longer provide a user friendly message (like `Option '--db' (vendor) expects a single value. Expected values are: ...` or `Option '--features' (feature) expects one or more comma separated values without whitespace. Expected values are: ...`) in these cases any more. However, I still do believe it's a step in the right direction as we're aligning with Picocli default behaviour and it allows us to remove custom code (`PropertyMapperParameterConsumer`).